### PR TITLE
doc(ticdc): add design document for storage sink

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: ref #6797

### What is changed and how it works?
1. add design document for storage sink

#### Tests 

 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: docs/design/2023-03-15-ticdc-storage-sink.md [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+### Functional Tests
+
+- Regular unit testing and integration testing cover the correctness of data replication using csv and canal-json protocol.
+- Manually test the availability and correctness of data synchronization using different external storage.
+
+### Scenario Tests
+
+Run stability and chaos tests under different workloads.
+  - The upstream and downstream data are consistent.
+  - Throughput and latency are stable for most scenario.
```
Comment (amyangfei) [inline docs/design/2023-03-15-ticdc-storage-sink.md:374]:
```suggestion
  - Throughput and latency are stable for most scenarios.
```

File: docs/design/2023-03-15-ticdc-storage-sink.md [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+- `partition-separator`: specifies the separator that separates the path by the table partition, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**20**</code>.
+- `date-separator`: classifies the files by the transaction commit date. Value options are:
+    - `none`: no `date-separator`. For example, all files with `test.table1` version being `9999` are saved to `s3://bucket/prefix1/prefix2/schema1/table1/9999`.
+    - `year`: the separator is the year of the transaction commit date, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**2022**</code>.
+    - `month`: the separator is the year and month of the transaction commit date, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**2022-01**</code>.
+    - `day`: the separator is the year, month, and day of the transaction commit date, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**2022-01-02**</code>.
+- `num`: saves the serial number of the file that records the data change, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/2022-01-02/CDC**000005**.csv</code>.
```
Comment (amyangfei) [inline docs/design/2023-03-15-ticdc-storage-sink.md:90]: When does the serial number increase, is there any restriction to the size of single csv file?

File: docs/design/2023-03-15-ticdc-storage-sink.md [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+}
+```
+
+### Protocols
+
+#### CSV
+
+##### Transactional constraints
+- In a single CSV file, the commit-ts of a row is equal to or smaller than that of the subsequent row.
+- The same transactions of a single table are stored in the same CSV file.
```
Comment (amyangfei) [inline docs/design/2023-03-15-ticdc-storage-sink.md:303]: Is this suitable for large transaction?

File: docs/design/2023-03-15-ticdc-storage-sink.md (hunk 2 above)
Comment (CharlesCheung96) [reply docs/design/2023-03-15-ticdc-storage-sink.md:90]: Data change files are saved to the downstream when either flush-interval or file-size meets the requirements. Ref: https://docs.pingcap.com/tidb/stable/ticdc-sink-to-cloud-storage#configure-nfs

File: docs/design/2023-03-15-ticdc-storage-sink.md (hunk 3 above)
Comment (CharlesCheung96) [reply docs/design/2023-03-15-ticdc-storage-sink.md:303]: Good advice, the above conditions can be guaranteed only when transaction-atomicity is set to table level.

Comment (CharlesCheung96) [inline docs/design/2023-03-15-ticdc-storage-sink.md:303]:
```suggestion
- The same transactions of a single table are stored in the same CSV file when `transaction-atomicity` is set to table level.
```

**Review by amyangfei — APPROVED** [review]

**Pull request merged**